pkg/newlog/cmd/vector.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2025 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"os"
+	"slices"
+	"strings"
+	"time"
+)
+
+var (
+	uploadSockVectorSource = "/run/devUpload_source.sock"
+	keepSockVectorSource   = "/run/devKeep_source.sock"
+	uploadSockVectorSink   = "/run/devUpload_sink.sock"
+	keepSockVectorSink     = "/run/devKeep_sink.sock"
+)
+
+type BufferedSockWriter struct {
+	path      string
+	buffer    chan []byte
+	reconnect time.Duration
+}
+
+// NewBufferedSockWriter creates a new buffered socket writer that writes to the specified path.
+// It uses a buffered channel to queue messages and attempts to reconnect if the connection is lost.
+func NewBufferedSockWriter(path string, bufSize int, reconnect time.Duration) *BufferedSockWriter {
+	sw := &BufferedSockWriter{
+		path:      path,
+		buffer:    make(chan []byte, bufSize),
+		reconnect: reconnect,
+	}
+	go sw.run()
+	return sw
+}
+
+func (sw *BufferedSockWriter) run() {
+	for {
+		conn, err := net.Dial("unix", sw.path)
+		if err != nil {
+			log.Errorf("socket connect failed: %v, retrying...", err)
+			time.Sleep(sw.reconnect)
+			continue
+		}
+
+		for msg := range sw.buffer {
+			_, err := conn.Write(msg)
+			if err != nil {
+				log.Errorf("socket write failed: %v, reconnecting...", err)
+				conn.Close()
+				break // reconnect
+			}
+		}
+	}
+}
+
+// Write implements the io.Writer interface for bufferedSockWriter.
+func (sw *BufferedSockWriter) Write(p []byte) (int, error) {
+	// Don't block forever, drop if buffer full
+	select {
+	case sw.buffer <- slices.Clone(p): // copy buffer
```

Comment:
I'm a bit worried about copying all the logs here, as it may introduce some overhead, but I hope we can profile it somehow in case something goes wrong.